services/users/src/main/java/org/opfab/users/services/UserSettingsService.java

```diff
@@ -61,7 +70,11 @@ public OperationResult<UserSettings> fetchUserSettings(String login) {
     }
 
     @SuppressWarnings("java:S2583") // false positive , it does not return always the same value as Sonar says 
-    public OperationResult<UserSettings> patchUserSettings(String login, UserSettings userSettingsPatch) {
+    public OperationResult<UserSettings> patchUserSettings(User user, String login, UserSettings userSettingsPatch) {
```

Comment:
To avoid confusion , i suggest to name the parameters more precisily 

Suggestion : User userRequestingPatch , String userLoginToPatch